### PR TITLE
Fix `global add` on Windows

### PR DESCRIPTION
**Pull request by FLGMwt**
**Summary**
`yarn global add <package>` is almost entirely broken on Windows due to the global linking workflow trying to use cmd-shim for linking more than one level. 😞 

During a global install, we first do the normal install, which cmd-shims all the bins into node_modules/.bin. cmd-shiming involves creating a `sh` style and `cmd` style passthru script for each bin in the package. However, since yarn globals link to those and not the package bins directly (like npm does), we need and extra level of linking from globalbin to node_modules/.bin.

`node_modules/.bin/pm2` turns into `node/pm2` and `node/pm2.cmd` 
then
`node_modules/.bin/pm2.cmd` turns into `node/pm2.cmd` and `node/pm2.cmd.cmd`

The second `node/pm2.cmd` overwrites the first, but there's no way to configure cmd-shim to do anything else. Whats more, neither of the two `pm2.cmd`s are valid. The first one is a cmd-style wrapper to a sh style script and the second (despite the name) is a sh-style script to a cmd style script in the /.bin.

The two valid files are `node/pm2` which is an sh that calls the `.bin/pm2` sh which calls the package bin and `node/pm2.cmd.cmd` which is a cmd that calls the `.bin/pm2.cmd` which calls the package bin. Without any way to configure cmd-shim, I opted for renaming `node/pm2.cmd.cmd` as `node/pm2.cmd` after the shims are done.

I've been struggling for a while about how to fix this (might involve a forking cmd-shim), but I think it's worth _temporarily_ shifting the jank from global installs not working at all to not working exactly as they should.

**Test plan**
Before: `yarn global add pm2` resulted in a broken global install and an extra file `pm2.cmd.cmd`. 

After changes, pm2 works correctly after installing in powershell shown here: 

```
> yarn global add pm2
yarn global v0.15.1
warning No license field
...
success Installed pm2@2.0.18 with binaries:
      - pm2
      - rundev
      - pm2-dev
      - pm2-docker
Done in 8.26s.
>
> pm2 -v
2.0.18
```

and the global bin has only the sh and cmd files:

```
> ls pm2*
    Directory: C:\Program Files\nodejs
Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a----       10/14/2016  12:06 AM            420 pm2
-a----       10/14/2016  12:06 AM            428 pm2-dev
-a----       10/14/2016  12:06 AM             97 pm2-dev.cmd
-a----       10/14/2016  12:06 AM            434 pm2-docker
-a----       10/14/2016  12:06 AM            100 pm2-docker.cmd
-a----       10/14/2016  12:06 AM             93 pm2.cmd
```

### CODE REVIEW

**Comment by FLGMwt** [comment]
Fixes #927 and #958

**Comment by onemen** [comment]
What do you think about #1226 ?

File: src/cli/commands/global.js [hunk 1]
```diff
@@ -114,6 +114,9 @@ async function initUpdateBins(config: Config, reporter: Reporter): Promise<() =>
       const dest = path.join(binFolder, path.basename(src));
       await fs.unlink(dest);
       await linkBin(src, dest);
+      if (process.platform === 'win32' && dest.indexOf('.cmd') != -1) {
```
Comment (bestander) [inline src/cli/commands/global.js]: nit: `!== -1`

**Comment by darahak** [comment]
@FLGMwt Are you still working on it? Looks like it's so close to being merged.

**Comment by leohxj** [comment]
merged? I am using 0.17.10 on Windows7, it still create .cmd.cmd file.

**Comment by FLGMwt** [comment]
Sorry all! Lost my GitHub notifications for awhile and missed the response.

Nit fixed : )

**Comment by timfish** [comment]
Has this made it into a release. I'm using v0.18.1 and global packages are not working.

**Pull request merged**